diff --git a/Makefile b/Makefile.patched
index 7ca93f0..8115d7a 100644
--- a/Makefile
+++ b/Makefile.patched
@@ -118,6 +118,13 @@ else
 LDEXPORT=-rdynamic
 endif
 
+# LDFLAGS for building dynamically-loadable JS binary modules
+LDFLAGS_DYNLOAD=$(LDFLAGS)
+ifdef CONFIG_DARWIN
+LDFLAGS_DYNLOAD += -undefined dynamic_lookup
+endif
+
+
 PROGS=qjs$(EXE) qjsbn$(EXE) qjsc qjsbnc run-test262 run-test262-bn
 ifndef CONFIG_WIN32
 PROGS+=qjscalc
@@ -368,19 +375,13 @@ doc/%.html: doc/%.texi
 ###############################################################################
 # tests
 
-ifndef CONFIG_DARWIN
-test: bjson.so
-endif
-
-test: qjs qjsbn
+test: qjs qjsbn bjson.so
 	./qjs tests/test_closure.js
 	./qjs tests/test_op.js
 	./qjs tests/test_builtin.js
 	./qjs tests/test_loop.js
 	./qjs -m tests/test_std.js
-ifndef CONFIG_DARWIN
 	./qjs -m tests/test_bjson.js
-endif
 	./qjsbn tests/test_closure.js
 	./qjsbn tests/test_op.js
 	./qjsbn tests/test_builtin.js
@@ -458,6 +459,6 @@ bench-v8: qjs qjs32
 	./qjs -d tests/bench-v8/combined.js
 
 bjson.so: $(OBJDIR)/bjson.pic.o
-	$(CC) $(LDFLAGS) -shared -o $@ $^ $(LIBS)
+	$(CC) $(LDFLAGS_DYNLOAD) -shared -o $@ $^ $(LIBS)
 
 -include $(wildcard $(OBJDIR)/*.d)
